Gem.install caused a BootstrapMethodError

**Environment Information**
  
Provide at least:
* JRuby version **9.3.3.0**
* Operating system and platform: **Happens on both Docker Ubuntu and OSX Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:41 PST 2021; root:xnu-8019.61.5~1/RELEASE_ARM64_T6000 arm64**

I'm using JRuby through org.jruby.embed.jsr223.JRubyEngineFactory, inside an osgi bundle

**Expected Behavior**

Calling `Gem.install(any_valid_installable_gem)` should install the gem if it isn't already installed, or proceed without error.

```java
private final javax.script.ScriptEngineFactory factory = new org.jruby.embed.jsr223.JRubyEngineFactory();
javax.script.ScriptEngine engine = factory.getScriptEngine();
engine.eval("Gem.install('any valid gem')");
```

**Actual Behavior**

Gem.install caused the following error:

## Error type 1
```
java.lang.BootstrapMethodError: bootstrap method initialization exception
        at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.stub_specification.RUBY$block$data$1(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/stub_specification.rb:118) ~[?:?]
        at org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:151) ~[?:?]
        at org.jruby.runtime.BlockBody.yield(BlockBody.java:106) ~[?:?]
        at org.jruby.runtime.Block.yield(Block.java:188) ~[?:?]
        at org.jruby.RubyIO.ensureYieldClose(RubyIO.java:1154) ~[?:?]
        at org.jruby.RubyIO.open(RubyIO.java:1148) ~[?:?]
        at org.jruby.RubyIO$INVOKER$s$0$0$open.call(RubyIO$INVOKER$s$0$0$open.gen) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:218) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:214) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:223) ~[?:?]
        at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.stub_specification.invokeOther12:open(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/stub_specification.rb:113) ~[?:?]
        at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.stub_specification.RUBY$method$data$0(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/stub_specification.rb:113) ~[?:?]
        at org.jruby.internal.runtime.methods.CompiledIRMethod.call(CompiledIRMethod.java:152) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:148) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:198) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:142) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:128) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:115) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:85) ~[?:?]
        at org.jruby.RubySymbol$SymbolProcBody.yieldInner(RubySymbol.java:1439) ~[?:?]
        at org.jruby.RubySymbol$SymbolProcBody.doYield(RubySymbol.java:1454) ~[?:?]
        at org.jruby.runtime.BlockBody.yield(BlockBody.java:108) ~[?:?]
        at org.jruby.runtime.Block.yield(Block.java:188) ~[?:?]
        at org.jruby.RubyArray.selectCommon(RubyArray.java:2712) ~[?:?]
        at org.jruby.RubyArray.select(RubyArray.java:2721) ~[?:?]
        at org.jruby.RubyArray$INVOKER$i$0$0$select.call(RubyArray$INVOKER$i$0$0$select.gen) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:152) ~[?:?]
        at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.specification.invokeOther2:select(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/specification.rb:774) ~[?:?]
        at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.specification.RUBY$method$gemspec_stubs_in$0(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/specification.rb:774) ~[?:?]
        at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.specification.RUBY$method$gemspec_stubs_in$0$__VARARGS__(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/specification.rb:773) ~[?:?]
        at org.jruby.internal.runtime.methods.CompiledIRMethod.call(CompiledIRMethod.java:139) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:112) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:85) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:94) ~[?:?]
        at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:546) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:361) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:86) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:201) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:188) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:206) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:173) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:316) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:92) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:238) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:225) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:214) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:204) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:325) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:80) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:164) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:151) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:198) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:142) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:80) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:164) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:151) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:198) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:142) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:80) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:164) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:151) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:152) ~[?:?]
        at org.jruby.RubyEnumerable.callEach(RubyEnumerable.java:87) ~[?:?]
        at org.jruby.RubyEnumerable.any_pCommon(RubyEnumerable.java:1742) ~[?:?]
        at org.jruby.RubyEnumerable.any_p(RubyEnumerable.java:1710) ~[?:?]
        at org.jruby.RubyEnumerable$INVOKER$s$any_p.call(RubyEnumerable$INVOKER$s$any_p.gen) ~[?:?]
        at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroOrOneBlock.call(JavaMethod.java:580) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:329) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:87) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:94) ~[?:?]
        at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:546) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:361) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:80) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:164) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:151) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:198) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:142) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:116) ~[?:?]
        at org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:136) ~[?:?]
        at org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:66) ~[?:?]
        at org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:58) ~[?:?]
        at org.jruby.runtime.Block.call(Block.java:143) ~[?:?]
        at org.jruby.RubyProc.call(RubyProc.java:309) ~[?:?]
        at org.jruby.internal.runtime.RubyRunnable.run(RubyRunnable.java:105) ~[?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
```

## Error type 2

Under certain situations it sometimes throw this error:
```
java.lang.BootstrapMethodError: bootstrap method initialization exception
        at java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:194) ~[?:?]
        at java.lang.invoke.CallSite.makeSite(CallSite.java:307) ~[?:?]
        at java.lang.invoke.MethodHandleNatives.linkCallSiteImpl(MethodHandleNatives.java:258) ~[?:?]
        at java.lang.invoke.MethodHandleNatives.linkCallSite(MethodHandleNatives.java:248) ~[?:?]
        at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.stub_specification.RUBY$block$data$1(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/stub_specification.rb:118) ~[?:?]
        at org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:151) ~[?:?]
        at org.jruby.runtime.BlockBody.yield(BlockBody.java:106) ~[?:?]
        at org.jruby.runtime.Block.yield(Block.java:188) ~[?:?]
        at org.jruby.RubyIO.ensureYieldClose(RubyIO.java:1154) ~[?:?]
        at org.jruby.RubyIO.open(RubyIO.java:1148) ~[?:?]
        at org.jruby.RubyIO$INVOKER$s$0$0$open.call(RubyIO$INVOKER$s$0$0$open.gen) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:218) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:404) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:216) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:223) ~[?:?]
        at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.stub_specification.invokeOther12:open(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/stub_specification.rb:113) ~[?:?]
        at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.stub_specification.RUBY$method$data$0(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/stub_specification.rb:113) ~[?:?]
        at org.jruby.internal.runtime.methods.CompiledIRMethod.call(CompiledIRMethod.java:152) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:148) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:198) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:351) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:144) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:128) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:115) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:329) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:87) ~[?:?]
        at org.jruby.RubySymbol$SymbolProcBody.yieldInner(RubySymbol.java:1439) ~[?:?]
        at org.jruby.RubySymbol$SymbolProcBody.doYield(RubySymbol.java:1454) ~[?:?]
        at org.jruby.runtime.BlockBody.yield(BlockBody.java:108) ~[?:?]
        at org.jruby.runtime.Block.yield(Block.java:188) ~[?:?]
        at org.jruby.RubyArray.selectCommon(RubyArray.java:2687) ~[?:?]
        at org.jruby.RubyArray.select(RubyArray.java:2721) ~[?:?]
        at org.jruby.RubyArray$INVOKER$i$0$0$select.call(RubyArray$INVOKER$i$0$0$select.gen) ~[?:?]
        at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroBlock.call(JavaMethod.java:560) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:85) ~[?:?]
        at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:549) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:361) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:128) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:115) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:85) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:94) ~[?:?]
        at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:546) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:361) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:116) ~[?:?]
        at org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:136) ~[?:?]
        at org.jruby.runtime.IRBlockBody.doYield(IRBlockBody.java:170) ~[?:?]
        at org.jruby.runtime.BlockBody.yield(BlockBody.java:108) ~[?:?]
        at org.jruby.runtime.Block.yield(Block.java:188) ~[?:?]
        at org.jruby.RubyEnumerable$19.call(RubyEnumerable.java:854) ~[?:?]
        at org.jruby.runtime.CallBlock.doYield(CallBlock.java:97) ~[?:?]
        at org.jruby.runtime.BlockBody.yield(BlockBody.java:108) ~[?:?]
        at org.jruby.runtime.Block.yield(Block.java:188) ~[?:?]
        at org.jruby.RubyArray.each(RubyArray.java:1865) ~[?:?]
        at org.jruby.RubyArray$INVOKER$i$0$0$each.call(RubyArray$INVOKER$i$0$0$each.gen) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:152) ~[?:?]
        at org.jruby.RubyEnumerable.callEach(RubyEnumerable.java:87) ~[?:?]
        at org.jruby.RubyEnumerable.flatMapCommon(RubyEnumerable.java:848) ~[?:?]
        at org.jruby.RubyEnumerable.flat_map(RubyEnumerable.java:836) ~[?:?]
        at org.jruby.RubyEnumerable$INVOKER$s$0$0$flat_map.call(RubyEnumerable$INVOKER$s$0$0$flat_map.gen) ~[?:?]
        at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroBlock.call(JavaMethod.java:560) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:85) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:94) ~[?:?]
        at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:546) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:361) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:128) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:115) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:85) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:94) ~[?:?]
        at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:546) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:361) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:92) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:238) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:225) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:214) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:204) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:325) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:86) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:201) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:188) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:206) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:173) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:316) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:86) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:201) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:188) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:206) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:173) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:316) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:80) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:164) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:151) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:198) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:142) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:86) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:201) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:188) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:206) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:173) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:316) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:86) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:201) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:188) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:206) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:173) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:316) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:116) ~[?:?]
        at org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:136) ~[?:?]
        at org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:66) ~[?:?]
        at org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:48) ~[?:?]
        at org.jruby.runtime.Block.call(Block.java:169) ~[?:?]
        at org.jruby.RubyEnumerable$EachWithIndex.call(RubyEnumerable.java:1098) ~[?:?]
        at org.jruby.runtime.BlockCallback.call(BlockCallback.java:54) ~[?:?]
        at org.jruby.runtime.CallBlock.doYield(CallBlock.java:102) ~[?:?]
        at org.jruby.runtime.BlockBody.yield(BlockBody.java:116) ~[?:?]
        at org.jruby.runtime.Block.yieldValues(Block.java:234) ~[?:?]
        at org.jruby.RubyEnumerator$1.call(RubyEnumerator.java:391) ~[?:?]
        at org.jruby.runtime.BlockCallback.call(BlockCallback.java:58) ~[?:?]
        at org.jruby.runtime.CallBlock.doYield(CallBlock.java:97) ~[?:?]
        at org.jruby.runtime.BlockBody.yield(BlockBody.java:108) ~[?:?]
        at org.jruby.runtime.Block.yield(Block.java:188) ~[?:?]
        at org.jruby.RubyEnumerable$10.doYield(RubyEnumerable.java:497) ~[?:?]
        at org.jruby.runtime.BlockBody.yield(BlockBody.java:108) ~[?:?]
        at org.jruby.runtime.Block.yield(Block.java:188) ~[?:?]
        at org.jruby.RubyArray.each(RubyArray.java:1865) ~[?:?]
        at org.jruby.RubyArray$INVOKER$i$0$0$each.call(RubyArray$INVOKER$i$0$0$each.gen) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:152) ~[?:?]
        at org.jruby.RubyEnumerable.each(RubyEnumerable.java:113) ~[?:?]
        at org.jruby.RubyEnumerable.sort_by(RubyEnumerable.java:487) ~[?:?]
        at org.jruby.RubyEnumerable$INVOKER$s$0$0$sort_by.call(RubyEnumerable$INVOKER$s$0$0$sort_by.gen) ~[?:?]
        at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroBlock.call(JavaMethod.java:560) ~[?:?]
        at org.jruby.RubyClass.finvokeWithRefinements(RubyClass.java:510) ~[?:?]
        at org.jruby.RubyClass.finvoke(RubyClass.java:498) ~[?:?]
        at org.jruby.runtime.Helpers.invoke(Helpers.java:645) ~[?:?]
        at org.jruby.RubyBasicObject.callMethod(RubyBasicObject.java:387) ~[?:?]
        at org.jruby.RubyEnumerator.__each__(RubyEnumerator.java:387) ~[?:?]
        at org.jruby.RubyEnumerator.each(RubyEnumerator.java:383) ~[?:?]
        at org.jruby.RubyEnumerator$INVOKER$i$each.call(RubyEnumerator$INVOKER$i$each.gen) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:152) ~[?:?]
        at org.jruby.RubyEnumerable.callEach(RubyEnumerable.java:95) ~[?:?]
        at org.jruby.RubyEnumerator.with_index_common(RubyEnumerator.java:562) ~[?:?]
        at org.jruby.RubyEnumerator.with_index(RubyEnumerator.java:572) ~[?:?]
        at org.jruby.RubyEnumerator$INVOKER$i$with_index.call(RubyEnumerator$INVOKER$i$with_index.gen) ~[?:?]
        at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroOrOneBlock.call(JavaMethod.java:580) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:85) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:94) ~[?:?]
        at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:546) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:361) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:128) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:115) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:85) ~[?:?]
        at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:549) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:361) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:116) ~[?:?]
        at org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:136) ~[?:?]
        at org.jruby.runtime.IRBlockBody.yieldSpecific(IRBlockBody.java:76) ~[?:?]
        at org.jruby.runtime.Block.yieldSpecific(Block.java:157) ~[?:?]
        at org.jruby.ir.targets.indy.YieldSite.yieldSpecific(YieldSite.java:161) ~[?:?]
        at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.resolver.molinillo.lib.molinillo.delegates.specification_provider.RUBY$method$with_no_such_dependency_error_handling$0(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:77) ~[?:?]
        at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.resolver.molinillo.lib.molinillo.delegates.specification_provider.RUBY$method$with_no_such_dependency_error_handling$0$__VARARGS__(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:76) ~[?:?]
        at org.jruby.internal.runtime.methods.CompiledIRMethod.call(CompiledIRMethod.java:139) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:112) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:329) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:87) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:94) ~[?:?]
        at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:546) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:361) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:128) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:115) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:329) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:87) ~[?:?]
        at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:549) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:361) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:92) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:238) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:225) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:214) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:393) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:206) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:325) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:86) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:201) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:188) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:206) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:372) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:175) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:316) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:80) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:164) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:151) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:198) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:351) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:144) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:80) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:164) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:151) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:198) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:351) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:144) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:80) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:164) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:151) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:198) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:351) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:144) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:80) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:164) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:151) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:198) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:351) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:144) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:92) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:238) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:225) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:214) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:393) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:206) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:325) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:80) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:164) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:151) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:198) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:351) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:144) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:86) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:201) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:188) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:206) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:372) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:175) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:316) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:92) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:238) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:225) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:214) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:393) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:206) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:325) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:92) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:238) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:225) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:214) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:393) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:206) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:325) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:86) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:201) ~[?:?]
        at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:188) ~[?:?]
        at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:206) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:372) ~[?:?]
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:175) ~[?:?]
        at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:316) ~[?:?]
        at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72) ~[?:?]
        at org.jruby.ir.interpreter.Interpreter.INTERPRET_ROOT(Interpreter.java:96) ~[?:?]
        at org.jruby.ir.interpreter.Interpreter.execute(Interpreter.java:81) ~[?:?]
        at org.jruby.ir.interpreter.Interpreter.execute(Interpreter.java:30) ~[?:?]
        at org.jruby.ir.IRTranslator.execute(IRTranslator.java:42) ~[?:?]
        at org.jruby.Ruby.runInterpreter(Ruby.java:1292) ~[?:?]
        at org.jruby.Ruby.runInterpreter(Ruby.java:1296) ~[?:?]
        at org.jruby.embed.internal.EmbedEvalUnitImpl.run(EmbedEvalUnitImpl.java:119) ~[?:?]
        at org.jruby.embed.jsr223.JRubyEngine.doEval(JRubyEngine.java:97) ~[?:?]
        at org.jruby.embed.jsr223.JRubyEngine.eval(JRubyEngine.java:124) ~[?:?]
        at org.jruby.embed.jsr223.JRubyEngine.eval(JRubyEngine.java:150) ~[?:?]
        at org.openhab.core.automation.module.script.internal.ScriptEngineManagerImpl.loadScript(ScriptEngineManagerImpl.java:177) ~[?:?]
        at org.openhab.core.automation.module.script.rulesupport.loader.ScriptFileWatcher.createAndLoad(ScriptFileWatcher.java:231) ~[?:?]
        at org.openhab.core.automation.module.script.rulesupport.loader.ScriptFileWatcher.importFile(ScriptFileWatcher.java:211) ~[?:?]
        at org.openhab.core.automation.module.script.rulesupport.loader.ScriptFileWatcher.lambda$2(ScriptFileWatcher.java:203) ~[?:?]
        at java.util.Optional.ifPresent(Optional.java:183) ~[?:?]
        at org.openhab.core.automation.module.script.rulesupport.loader.ScriptFileWatcher.importFileWhenReady(ScriptFileWatcher.java:201) ~[?:?]
        at org.openhab.core.automation.module.script.rulesupport.loader.ScriptFileWatcher.processWatchEvent(ScriptFileWatcher.java:178) ~[?:?]
        at org.openhab.core.service.WatchQueueReader.lambda$3(WatchQueueReader.java:323) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.RuntimeException: java.lang.IllegalAccessException: no such method: org.jruby.runtime.Helpers.constructRubyStringArray(RubyString,RubyString,RubyString)RubyString[]/invokeStatic
        at org.jruby.runtime.Helpers.constructRubyStringArrayHandle(Helpers.java:1510) ~[?:?]
        at org.jruby.ir.targets.indy.DRegexpObjectSite.prepareBinder(DRegexpObjectSite.java:64) ~[?:?]
        at org.jruby.ir.targets.indy.ConstructObjectSite.bootstrap(ConstructObjectSite.java:22) ~[?:?]
        at org.jruby.ir.targets.indy.DRegexpObjectSite.bootstrap(DRegexpObjectSite.java:43) ~[?:?]
        at java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:215) ~[?:?]
        at java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:102) ~[?:?]
        ... 308 more
Caused by: java.lang.IllegalAccessException: no such method: org.jruby.runtime.Helpers.constructRubyStringArray(RubyString,RubyString,RubyString)RubyString[]/invokeStatic
        at java.lang.invoke.MemberName.makeAccessException(MemberName.java:959) ~[?:?]
        at java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:1101) ~[?:?]
        at java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:2030) ~[?:?]
        at java.lang.invoke.MethodHandles$Lookup.findStatic(MethodHandles.java:1102) ~[?:?]
        at org.jruby.runtime.Helpers.constructRubyStringArrayHandle(Helpers.java:1505) ~[?:?]
        at org.jruby.ir.targets.indy.DRegexpObjectSite.prepareBinder(DRegexpObjectSite.java:64) ~[?:?]
        at org.jruby.ir.targets.indy.ConstructObjectSite.bootstrap(ConstructObjectSite.java:22) ~[?:?]
        at org.jruby.ir.targets.indy.DRegexpObjectSite.bootstrap(DRegexpObjectSite.java:43) ~[?:?]
        at java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:215) ~[?:?]
        at java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:102) ~[?:?]
        ... 308 more
Caused by: java.lang.LinkageError: loader constraint violation: when resolving method 'org.jruby.RubyString[] org.jruby.runtime.Helpers.constructRubyStringArray(org.jruby.RubyString, org.jruby.RubyString, org.jruby.RubyString)' the class loader 'bootstrap' of the current class, java/lang/Object, and the class loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @2fef16e1 for the method's defining class, org/jruby/runtime/Helpers, have different Class objects for the type org/jruby/RubyString used in the signature (java.lang.Object is in module java.base of loader 'bootstrap'; org.jruby.runtime.Helpers is in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @2fef16e1, parent loader java.net.URLClassLoader @9a7504c)
        at java.lang.invoke.MethodHandleNatives.resolve(Native Method) ~[?:?]
        at java.lang.invoke.MemberName$Factory.resolve(MemberName.java:1070) ~[?:?]
        at java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:1098) ~[?:?]
        at java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:2030) ~[?:?]
        at java.lang.invoke.MethodHandles$Lookup.findStatic(MethodHandles.java:1102) ~[?:?]
        at org.jruby.runtime.Helpers.constructRubyStringArrayHandle(Helpers.java:1505) ~[?:?]
        at org.jruby.ir.targets.indy.DRegexpObjectSite.prepareBinder(DRegexpObjectSite.java:64) ~[?:?]
        at org.jruby.ir.targets.indy.ConstructObjectSite.bootstrap(ConstructObjectSite.java:22) ~[?:?]
        at org.jruby.ir.targets.indy.DRegexpObjectSite.bootstrap(DRegexpObjectSite.java:43) ~[?:?]
        at java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:215) ~[?:?]
        at java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:102) ~[?:?]
        ... 308 more

```

@jimtng Sometimes I see a similar error:
```
Caused by: java.lang.LinkageError: loader constraint violation: when resolving method 'org.jruby.RubyString[] org.jruby.runtime.Helpers.constructRubyStringArray(org.jruby.RubyString, org.jruby.RubyString, org.jruby.RubyString, org.jruby.RubyString)' the class loader 'bootstrap' of the current class, java/lang/Object, and the class loader org.apache.felix.framework.BundleWiringImpl$BundleClassLoader @58634468 for the method's defining class, org/jruby/runtime/Helpers, have different Class objects for the type org/jruby/RubyString used in the signature (java.lang.Object is in module java.base of loader 'bootstrap'; org.jruby.runtime.Helpers is in unnamed module of loader org.apache.felix.framework.BundleWiringImpl$BundleClassLoader @58634468, parent loader org.apache.catalina.loader.ParallelWebappClassLoader @7957dc72)
```

It seems that one of the causes is that after I uninstall this OSGi plugin there are still many JRuby objects remaining in the memory and I have not yet debugged what keeps the references to them. And then during the next installation of the plugin somehow the old existing JRuby java classes are found which causes this error about different class loader.

--- a/core/src/main/java/org/jruby/runtime/Helpers.java
+++ b/core/src/main/java/org/jruby/runtime/Helpers.java
@@ -1436,7 +1436,7 @@ public class Helpers {
                 if (size == 0) {
                     handle = Binder.from(IRubyObject[].class).getStatic(LOOKUP, IRubyObject.class, "NULL_ARRAY");
                 } else {
-                    handle = MethodHandles.publicLookup().findStatic(Helpers.class, "constructObjectArray", MethodType.methodType(IRubyObject[].class, CodegenUtils.params(IRubyObject.class, size)));
+                    handle = MethodHandles.lookup().findStatic(Helpers.class, "constructObjectArray", MethodType.methodType(IRubyObject[].class, CodegenUtils.params(IRubyObject.class, size)));
                 }
 
                 constructObjectArrayHandles[size] = handle;
@@ -1506,7 +1506,7 @@ public class Helpers {
                 if (size == 0) {
                     handle = Binder.from(RubyString[].class).getStatic(LOOKUP, RubyString.class, "NULL_ARRAY");
                 } else {
-                    handle = MethodHandles.publicLookup().findStatic(Helpers.class, "constructRubyStringArray", MethodType.methodType(RubyString[].class, CodegenUtils.params(RubyString.class, size)));
+                    handle = MethodHandles.lookup().findStatic(Helpers.class, "constructRubyStringArray", MethodType.methodType(RubyString[].class, CodegenUtils.params(RubyString.class, size)));
                 }
 
                 constructRubyStringArrayHandles[size] = handle;

